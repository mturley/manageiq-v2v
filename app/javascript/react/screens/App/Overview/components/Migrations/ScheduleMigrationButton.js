import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'patternfly-react';

const ScheduleMigrationButton = ({ loading, toggleScheduleMigrationModal, plan, isMissingMapping }) => {
  const retry = plan.status === 'Error' || plan.status === 'Denied';

  return (
    <Button
      id={`schedule_${plan.id}`}
      onClick={e => {
        e.stopPropagation();
        toggleScheduleMigrationModal({ plan });
      }}
      disabled={isMissingMapping || loading === plan.href || plan.schedule_type}
    >
      {retry ? __('Schedule Retry') : __('Schedule Migration')}
    </Button>
  );
};

ScheduleMigrationButton.propTypes = {
  loading: PropTypes.string,
  toggleScheduleMigrationModal: PropTypes.func,
  plan: PropTypes.object,
  isMissingMapping: PropTypes.bool
};

export default ScheduleMigrationButton;
