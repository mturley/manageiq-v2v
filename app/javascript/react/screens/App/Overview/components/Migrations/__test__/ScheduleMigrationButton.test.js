import React from 'react';
import { mount, shallow } from 'enzyme';
import ScheduleMigrationButton from '../ScheduleMigrationButton';

describe('When initialized with cold migration plan', () => {
  const plan = {
    options: { id: 1, status: 'Ok', href: 'url', config_info: { warm_migration: false } }
  };
  const baseProps = {
    plan,
    loading: false,
    isMissingMapping: false,
    toggleScheduleMigrationModal: jest.fn(() => Promise.resolve())
  };

  test('renders Schedule Migration button', () => {
    const wrapper = shallow(<ScheduleMigrationButton {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders Schedule Retry button', () => {
    baseProps.plan.status = 'Error';
    const wrapper = shallow(<ScheduleMigrationButton {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('clicking the schedule button launches modal window', async () => {
    const wrapper = mount(<ScheduleMigrationButton {...baseProps} />);
    wrapper.find('button').simulate('click');
    await expect(baseProps.toggleScheduleMigrationModal).toHaveBeenCalledWith({ plan });
  });
});

describe('When initialized with warm migration plan', () => {
  const plan = {
    options: { id: 1, status: 'Ok', href: 'url', config_info: { warm_migration: true } }
  };
  const baseProps = {
    plan,
    loading: false,
    isMissingMapping: false,
    toggleScheduleMigrationModal: jest.fn(() => Promise.resolve())
  };

  test('renders Schedule Cutover button', () => {
    const wrapper = shallow(<ScheduleMigrationButton {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders Schedule Retry button', () => {
    baseProps.plan.status = 'Error';
    const wrapper = shallow(<ScheduleMigrationButton {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('clicking the schedule button launches modal window', async () => {
    const wrapper = mount(<ScheduleMigrationButton {...baseProps} />);
    wrapper.find('button').simulate('click');
    await expect(baseProps.toggleScheduleMigrationModal).toHaveBeenCalledWith({ plan });
  });
});
