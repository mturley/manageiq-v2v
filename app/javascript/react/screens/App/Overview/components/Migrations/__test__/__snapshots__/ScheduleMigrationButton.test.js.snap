// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When initialized with cold migration plan renders Schedule Migration button 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  id="schedule_undefined"
  onClick={[Function]}
>
  Schedule Migration
</Button>
`;

exports[`When initialized with cold migration plan renders Schedule Retry button 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  id="schedule_undefined"
  onClick={[Function]}
>
  Schedule Retry
</Button>
`;

exports[`When initialized with warm migration plan renders Schedule Cutover button 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  id="schedule_undefined"
  onClick={[Function]}
>
  Schedule Migration
</Button>
`;

exports[`When initialized with warm migration plan renders Schedule Retry button 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  id="schedule_undefined"
  onClick={[Function]}
>
  Schedule Retry
</Button>
`;
